Vet clinic database: add "join table" for visits

M1rn4 opened the pull request:
# Vet clinic database: add "join table" for visits
-  Understand the different types of relationships between tables.
-  Prepare complex queries that answer analytical questions.
-  Use primary key & foreign key mechanisms for joining tables.

# Write queries to answer the following:

## 1.- Who was the last animal seen by William Thatcher?
     last_animal_seen_by_william_tatcher 
    -------------------------------------
     Blossom
    (1 row)

## 2.- How many different animals did Stephanie Mendez see?
  1
## 3.- List all vets and their specialties, including vets with no specialties.
         vet_name     | specialty_name 
    ------------------+----------------
     Jack Harkness    | 
     Maisy Smith      | 
     Stephanie Mendez | 
     William Tatcher  | 
    (4 rows)

## 4.- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
 animal_id | animal_name | visit_date 
-----------+-------------+------------
(0 rows)

## 5.- What animal has the most visits to vets?
     animal_name | num_visits 
    -------------+------------
     Boarmon     |          4
    (1 row)

## 6.- Who was Maisy Smith's first visit?
   first_visit 

## 7.- Details for most recent visit: animal information, vet information, and date of visit.
![Captura de pantalla de 2023-04-03 16-50-10](https://user-images.githubusercontent.com/81875032/229636524-697eb874-327a-4aa2-a83c-f025e90cf961.png)

## 8.- How many visits were with a vet that did not specialize in that animal's species?
0
## 9.- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
0
![Captura de pantalla de 2023-04-04 10-18-33](https://user-images.githubusercontent.com/81875032/229839724-8a0da53a-8420-4a7b-a575-274e6c805a2d.png)

![Captura de pantalla de 2023-04-03 16-50-51](https://user-images.githubusercontent.com/81875032/229636499-61f9516b-e686-45c9-8ff4-ee40a543325d.png)

![Captura de pantalla de 2023-04-03 16-50-48](https://user-images.githubusercontent.com/81875032/229636511-02648c53-5857-47b8-8e0c-d43c134bc551.png)

![Captura de pantalla de 2023-04-03 16-50-39](https://user-images.githubusercontent.com/81875032/229636517-30da82d8-0d33-4465-a02b-bcd040aea883.png)

M1rn4 committed:
join_table

M1rn4 committed:
Merge branch 'features' of github.com:M1rn4/Vet-Clinic- into features

rloterh reviewed (commented):
Hi Mirna, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project that appears to be under the wrong module/ activity. Your `queries.sql` file contains queries other than those required for **the Vet clinic database: add 'join table' for visits** module. Kindly ensure to resubmit this project under the right activity/ module.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

M1rn4 commented:
Dear Robert,

Thank you for taking the time to review my code on GitHub. I would like to address your comment regarding the queries used, which you mentioned are not correct.

After carefully reviewing my code, I have verified that the queries I used are correct and efficient for the purpose of my code. Although I understand that there may be different ways to write queries to achieve the same result, I am confident that the queries I used are appropriate and accurately respond to the questions requested on the page.

I appreciate your feedback and revisions, and I am available to answer any further questions or concerns you may have.

Best regards,

addod19 commented on schema.sql:
```diff
@@ -32,3 +32,21 @@ ALTER TABLE animals ADD COLUMN species_id INTEGER REFERENCES species(id);
+    visit_date DATE
+);
+
+CREATE TABLE specialties (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255) UNIQUE
+);
```
- [ ] It would be better if our `specialties` or `specialization` table contains the `species_id` and `vets_id` as column names to serve as a join table between the `species` and `vets` tables
![Screenshot from 2023-04-03 23-44-29](https://user-images.githubusercontent.com/46388113/229650300-fc11e6d3-cf7b-41e5-9cfa-0f65775f89e1.png)

addod19 commented on data.sql:
```diff
@@ -58,3 +58,39 @@ SET owner_id = (
+Boarmon: 9
+Blossom: 10
+Las consultas actualizadas serían las siguientes:
+
```
- [ ] It would be better if we add some data to our vets table as seen in the screenshot below
![Screenshot from 2023-04-03 23-49-38](https://user-images.githubusercontent.com/46388113/229650865-f6c107fc-5de7-4ba2-b801-2080c6bbfb91.png)

addod19 commented on data.sql:
```diff
@@ -58,3 +58,39 @@ SET owner_id = (
 )
 WHERE name IN ('Angemon', 'Boarmon');
 
```
- [ ] We are also required to add data into our `specialization` table or `specialties` as you've named it
![Screenshot from 2023-04-03 23-51-26](https://user-images.githubusercontent.com/46388113/229651185-82364422-6d7e-4583-9126-8075565c8f15.png)

rloterh replied:
> Dear Robert,
> 
> Thank you for taking the time to review my code on GitHub. I would like to address your comment regarding the queries used, which you mentioned are not correct.
> 
> After carefully reviewing my code, I have verified that the queries I used are correct and efficient for the purpose of my code. Although I understand that there may be different ways to write queries to achieve the same result, I am confident that the queries I used are appropriate and accurately respond to the questions requested on the page.
> 
> I appreciate your feedback and revisions, and I am available to answer any further questions or concerns you may have.
> 
> Best regards,

Hi @M1rn4 

I noticed that your screenshot had answers to the required queries, however, the required answers where missing in the `queries.sql` file, which was intended to hold your queries.

addod19 reviewed (changes requested):
### Hi  @M1rn4, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- [ ] Correct Git-flow followed :+1: 
- [ ] README is professional :100: 


## Required Changes ♻️
_Check the comments under the review._
- [ ] It would be better if our queries can answer the following questions as seen below
![Screenshot from 2023-04-03 23-54-42](https://user-images.githubusercontent.com/46388113/229651856-96e0de74-577d-4d2b-b933-f5889a91864e.png)
![Screenshot from 2023-04-03 23-54-53](https://user-images.githubusercontent.com/46388113/229651865-8dbc452c-a430-4e2d-949f-e57bc62e1135.png)
![Screenshot from 2023-04-03 23-55-09](https://user-images.githubusercontent.com/46388113/229651868-a4e35b06-5bfe-4f42-89a0-f49f027c95c1.png)
![Screenshot from 2023-04-03 23-55-26](https://user-images.githubusercontent.com/46388113/229651869-9cca56b6-058a-4620-8a54-94f2ed742938.png)
![Screenshot from 2023-04-03 23-55-44](https://user-images.githubusercontent.com/46388113/229651872-30984efe-7020-4060-a26c-b0e96928af3d.png)
![Screenshot from 2023-04-03 23-55-53](https://user-images.githubusercontent.com/46388113/229651877-9eca5415-9264-458a-b869-076a28593e40.png)
![Screenshot from 2023-04-03 23-56-11](https://user-images.githubusercontent.com/46388113/229651879-66672570-92db-4a86-8fbb-734db1baba4b.png)
![Screenshot from 2023-04-03 23-56-34](https://user-images.githubusercontent.com/46388113/229651884-cc89207d-7879-4fb3-a610-88628467d3c8.png)


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

M1rn4 committed:
changes

M1rn4 committed:
add changes

M1rn4 commented:
I sincerely appreciate you taking the time to review my code on GitHub and providing your valuable feedback. I want to apologize for the previous oversight of not pasting the code, but I want to let you know that I have updated both the code and queries, and they can now be fully observed.

I want to confirm that I have implemented all the suggested changes and carefully reviewed the code to ensure it meets the required standards. Thank you for your thorough review and for helping me improve my work.

Best regards,

addod19 reviewed (approved):
### Hi @M1rn4, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job addressing all the comments :rocket: awesome


### Optional
- [ ] Going forward, let us try to capitalize all our commit messages, for example; `Add changes` instead of `add changes`, etc

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
